test/UnitTestHelpers.hs

```diff
@@ -0,0 +1,352 @@
+import           Game.LambdaHack.Client.UI.Frontend
+import           Game.LambdaHack.Client.UI.Key (KMP (..))
+import qualified Game.LambdaHack.Client.UI.Key as K
+import           Game.LambdaHack.Client.UI.PointUI
+import           Game.LambdaHack.Client.UI.UIOptions
+
+import           Game.LambdaHack.Common.Actor
+import           Game.LambdaHack.Common.Area
+import           Game.LambdaHack.Common.ClientOptions
+import           Game.LambdaHack.Common.Faction
+import           Game.LambdaHack.Common.Kind
+import           Game.LambdaHack.Common.Level
+import           Game.LambdaHack.Common.MonadStateRead
+import           Game.LambdaHack.Common.Perception
+import           Game.LambdaHack.Common.Point
+import           Game.LambdaHack.Common.State
+import           Game.LambdaHack.Common.Time
+import           Game.LambdaHack.Common.Types
+
+import           Game.LambdaHack.Content.ItemKind
+import           Game.LambdaHack.Content.RuleKind
+import           Game.LambdaHack.Content.TileKind
+import qualified Game.LambdaHack.Core.Dice as Dice
+
+import qualified Game.LambdaHack.Definition.Ability as Ability
+import           Game.LambdaHack.Definition.Color
+import           Game.LambdaHack.Definition.Flavour
+
+import Content.FactionKind
+
+import Game.LambdaHack.Common.Misc
+
+
+-- Read UI requests from the client and send them to the frontend,
+fchanFrontendStub :: ChanFrontend
+fchanFrontendStub =
+  ChanFrontend $ \case
+    FrontFrame _ -> putStr "FrontFrame"
+    FrontDelay _ -> putStr "FrontDelay"
+    FrontKey _ _ -> return KMP { kmpKeyMod=K.escKM, kmpPointer=PointUI 0 0 }
+    FrontPressed -> return False
+    FrontDiscardKey -> putStr "FrontDiscardKey"
+    FrontResetKeys -> putStr "FrontResetKeys"
+    FrontShutdown -> putStr "FrontShutdown"
+    FrontPrintScreen -> putStr "FrontPrintScreen"
+
+data CliState = CliState
+  { cliState   :: State            -- ^ current global state
+  , cliClient  :: StateClient      -- ^ current client state
+  , cliSession :: Maybe SessionUI  -- ^ UI state, empty for AI clients
+  -- , cliDict    :: ChanServer       -- ^ this client connection information
+  -- , cliToSave  :: Save.ChanSave (StateClient, Maybe SessionUI)
+  }
+
+
+stubUIOptions :: UIOptions
+stubUIOptions = UIOptions
+  { uCommands = []
+  , uHeroNames = []
+  , uVi = False
+  , uLeftHand = False
+  , uChosenFontset = ""
+  , uAllFontsScale = 0.0
+  , uFullscreenMode = NotFullscreen      
+  , uhpWarningPercent = 0
+  , uMsgWrapColumn = 0
+  , uHistoryMax = 0
+  , uMaxFps = 0.0
+  , uNoAnim = False
+  , uOverrideCmdline = []
+  , uFonts = []
+  , uFontsets = []
+  , uMessageColors = []
+  }
+
+stubClientOptions :: ClientOptions
+stubClientOptions = defClientOptions 
+  { schosenFontset = Just "snoopy"
+  , sfontsets = [("snoopy",FontSet {fontMapScalable="scalable",fontMapBitmap="bitmap",fontPropRegular="propRegular",fontPropBold="propBold",fontMono="mono"})]
+  }
+
+testLevelDimension :: Int
```

Comment:
This needs to be at least 5, according to the latest validation of `RuleContent` (because level max dimentions must match screen size; we don't do scrolling or always blank map space).

Morever, if both dimentions are 5, my commit removing override of `coscreen` component of `emptyCCUI` was wrong, because we need to override `rwidth` to `testLevelDimension` and `rheight` to `testLevelDimension + 3` to end up with valid content. Sorry about that.